MavenBuildSupportTest.testDownloadSources fails

See https://ci.eclipse.org/ls/job/jdt-ls-master/723/
```
java.lang.AssertionError: Couldn't find source for org.apache.commons.lang3.StringUtils(/home/jenkins/.m2/repository/org/apache/commons/commons-lang3/3.5/commons-lang3-3.5-sources.jar is missing)
	at org.eclipse.jdt.ls.core.internal.managers.MavenBuildSupportTest.testDownloadSources(MavenBuildSupportTest.java:177)
```

Some cases related with keyword commons.lang3 also randomly fail. 
```
java.lang.AssertionError: No symbols found for org.apache.commons.lang3.text.StrTokenizer
	at org.eclipse.jdt.ls.core.internal.handlers.DocumentSymbolHandlerTest.getSymbols(DocumentSymbolHandlerTest.java:247)
	at org.eclipse.jdt.ls.core.internal.handlers.DocumentSymbolHandlerTest.testSyntheticMember(DocumentSymbolHandlerTest.java:76)
```

```
java.lang.AssertionError: No symbols found for org.apache.commons.lang3.text.WordUtils
	at org.eclipse.jdt.ls.core.internal.handlers.DocumentSymbolHandlerTest.getSymbols(DocumentSymbolHandlerTest.java:247)
	at org.eclipse.jdt.ls.core.internal.handlers.DocumentSymbolHandlerTest.testClass(DocumentSymbolHandlerTest.java:219)
	at org.eclipse.jdt.ls.core.internal.handlers.DocumentSymbolHandlerTest.testDocumentSymbolHandler(DocumentSymbolHandlerTest.java:65)
```

```
java.lang.AssertionError
	at org.eclipse.jdt.ls.core.internal.handlers.NavigateToTypeDefinitionHandlerTest.testClass(NavigateToTypeDefinitionHandlerTest.java:88)
	at org.eclipse.jdt.ls.core.internal.handlers.NavigateToTypeDefinitionHandlerTest.testAttachedSource(NavigateToTypeDefinitionHandlerTest.java:56)
```
See https://ci.eclipse.org/ls/job/jdt-ls-pr/1600/ and https://ci.eclipse.org/ls/job/jdt-ls-pr/1601/

Is it because of the inconsistency of `M2_HOME` ?

`/home/jenkins/.m2` is not writable, thus now the M2_HOME is:
```
11:46:27 + export 'MAVEN_USER_HOME=/home/jenkins/agent/workspace/jdt-ls-pr/.m2'
```

In the error log, it seems to look for lib from `/home/jenkins/.m2` 
```
java.lang.AssertionError: Couldn't find source for org.apache.commons.lang3.StringUtils
(/home/jenkins/.m2/repository/org/apache/commons/commons-lang3/3.5/commons-lang3-3.5-sources.jar is missing)
	at org.eclipse.jdt.ls.core.internal.managers.MavenBuildSupportTest.testDownloadSources(MavenBuildSupportTest.java:177)
```

--- a/org.eclipse.jdt.ls.tests/src/org/eclipse/jdt/ls/core/internal/managers/AbstractProjectsManagerBasedTest.java
+++ b/org.eclipse.jdt.ls.tests/src/org/eclipse/jdt/ls/core/internal/managers/AbstractProjectsManagerBasedTest.java
@@ -124,6 +124,7 @@ public abstract class AbstractProjectsManagerBasedTest {
 		preferences = new Preferences();
 		preferences.setRootPaths(Collections.singleton(new Path(getWorkingProjectDirectory().getAbsolutePath())));
 		preferences.setCodeGenerationTemplateGenerateComments(true);
+		preferences.setMavenDownloadSources(true);
 		if (preferenceManager == null) {
 			preferenceManager = mock(PreferenceManager.class);
 		}

--- a/org.eclipse.jdt.ls.tests/src/org/eclipse/jdt/ls/core/internal/managers/MavenBuildSupportTest.java
+++ b/org.eclipse.jdt.ls.tests/src/org/eclipse/jdt/ls/core/internal/managers/MavenBuildSupportTest.java
@@ -162,19 +162,25 @@ public class MavenBuildSupportTest extends AbstractMavenBasedTest {
 	public void testDownloadSources() throws Exception {
 		File file = DependencyUtil.getSources("org.apache.commons", "commons-lang3", "3.5");
 		FileUtils.deleteDirectory(file.getParentFile());
-		IProject project = importMavenProject("salut");
-		waitForBackgroundJobs();
-		assertTrue(!file.exists());
-		IJavaProject javaProject = JavaCore.create(project);
-		IType type = javaProject.findType("org.apache.commons.lang3.StringUtils");
-		IClassFile classFile = ((BinaryType) type).getClassFile();
-		assertNull(classFile.getBuffer());
-		String source = new SourceContentProvider().getSource(classFile, new NullProgressMonitor());
-		if (source == null) {
-			JobHelpers.waitForDownloadSourcesJobs(JobHelpers.MAX_TIME_MILLIS);
-			source = new SourceContentProvider().getSource(classFile, new NullProgressMonitor());
+		boolean mavenDownloadSources = preferences.isMavenDownloadSources();
+		try {
+			preferences.setMavenDownloadSources(false);
+			IProject project = importMavenProject("salut");
+			waitForBackgroundJobs();
+			assertTrue(!file.exists());
+			IJavaProject javaProject = JavaCore.create(project);
+			IType type = javaProject.findType("org.apache.commons.lang3.StringUtils");
+			IClassFile classFile = ((BinaryType) type).getClassFile();
+			assertNull(classFile.getBuffer());
+			String source = new SourceContentProvider().getSource(classFile, new NullProgressMonitor());
+			if (source == null) {
+				JobHelpers.waitForDownloadSourcesJobs(JobHelpers.MAX_TIME_MILLIS);
+				source = new SourceContentProvider().getSource(classFile, new NullProgressMonitor());
+			}
+			assertNotNull("Couldn't find source for " + type.getFullyQualifiedName() + "(" + file.getAbsolutePath() + (file.exists() ? " exists)" : " is missing)"), source);
+		} finally {
+			preferences.setMavenDownloadSources(mavenDownloadSources);
 		}
-		assertNotNull("Couldn't find source for " + type.getFullyQualifiedName() + "(" + file.getAbsolutePath() + (file.exists() ? " exists)" : " is missing)"), source);
 	}
 
 	@Test

